layer/mounted_layer.go

```diff
@@ -146,6 +156,13 @@ func (rl *referencedRWLayer) Unmount() error {
 		return ErrLayerNotRetained
 	}
 	rl.activityCount--
+	if rl.activityCount > 0 {
+		return ErrActiveMount
```

Comment:
Is the intention that this error would be ignored by the caller? Seems odd that increasing the activity beyond 1 can be done without throwing an error but decreasing it back down to one would involve handing this error. What information does the caller need about the mount state after calling this function, is it relevant that it is still mounted or not? ...assuming the caller is properly calling an Unmount for every corresponding Mount.